daffodil-core/src/main/scala/org/apache/daffodil/core/runtime1/TermRuntime1Mixin.scala

```diff
@@ -354,8 +355,12 @@ trait TermRuntime1Mixin { self: Term =>
           case (Closed(pnes1), Closed(pnes2)) => {
             thisItself // When everything is closed, we only want thisItself and not the entire list of Closed possibilities
           }
-          case (Open(pnes1), Closed(pnes2)) => {
+          case (Open(pnes1), Closed(pnes2)) if !this.isInstanceOf[ChoiceGroupRef] => {
             // If there are optional elements before a required element, the whole sequence is considered closed
+            //
+            // If this is a ChoiceGroupRef though, it could be one of multiple group references and thus we may
+            // not have the appropriate context to find the expected following element, so we should remain Open
```

Comment:
This comment says "so we should remain open", but the next line of code is a Closed.